### PR TITLE
[libxdf] Add new port

### PR DESCRIPTION
**Pull request by myd7349**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [x] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html)
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [x] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by JonLiu1993** [comment]
@myd7349, Thanks for your PR, if this PR is ready for review please let me know.

**Comment by JonLiu1993** [comment]
When I tested the usage, I get this error;

CMakeLists.txt
````
cmake_minimum_required (VERSION 3.8)

project(test)

find_package(libxdf CONFIG REQUIRED)

# Add source to this project's executable.
add_executable (test "usageTest.cpp")

target_link_libraries(test PRIVATE XDF::xdf)
````
usageTest.cpp
````
#include <stdio.h>
#include <xdf.h>

int main()
{
    printf("%s\n", "HELLO WORLD");
    return 0;
}
````
Config:
````
1> CMake generation started for configuration: 'x64-Debug'.
1> Command line: "C:\WINDOWS\system32\cmd.exe" /c "%SYSTEMROOT%\System32\chcp.com 65001 >NUL && "C:\PROGRAM FILES\MICROSOFT VISUAL STUDIO\2022\COMMUNITY\COMMON7\IDE\COMMONEXTENSIONS\MICROSOFT\CMAKE\CMake\bin\cmake.exe"  -G "Visual Studio 17 2022" -A x64  -DCMAKE_CONFIGURATION_TYPES:STRING="Debug" -DCMAKE_INSTALL_PREFIX:PATH="C:\Users\test\source\repos\usageTest\out\install\x64-Debug" -DCMAKE_TOOLCHAIN_FILE=F:/Feature-test/libxdf/vcpkg/scripts/buildsystems/vcpkg.cmake "C:\Users\test\source\repos\usageTest" 2>&1"
1> Working directory: C:\Users\test\source\repos\usageTest\out\build\x64-Debug
1> [CMake] -- Selecting Windows SDK version 10.0.22621.0 to target Windows 10.0.19045.
1> [CMake] -- The C compiler identification is MSVC 19.36.32537.0
1> [CMake] -- The CXX compiler identification is MSVC 19.36.32537.0
1> [CMake] -- Detecting C compiler ABI info
1> [CMake] -- Detecting C compiler ABI info - done
1> [CMake] -- Check for working C compiler: C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.36.32532/bin/Hostx64/x64/cl.exe - skipped
1> [CMake] -- Detecting C compile features
1> [CMake] -- Detecting C compile features - done
1> [CMake] -- Detecting CXX compiler ABI info
1> [CMake] -- Detecting CXX compiler ABI info - done
1> [CMake] -- Check for working CXX compiler: C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.36.32532/bin/Hostx64/x64/cl.exe - skipped
1> [CMake] -- Detecting CXX compile features
1> [CMake] -- Detecting CXX compile features - done
1> [CMake] -- Configuring done (7.5s)
1> [CMake] -- Generating done (0.0s)
1> [CMake] -- Build files have been written to: C:/Users/test/source/repos/usageTest/out/build/x64-debug
1> Extracted CMake variables.
1> Extracted source files and headers.
1> Extracted code model.
1> Extracted toolchain configurations.
1> Extracted includes paths.
1> CMake generation finished.

````
Build
````
>------ Build All started: Project: usageTest, Configuration: x64-Debug ------
  MSBuild version 17.6.3+07e294721 for .NET Framework
  
    Checking Build System
    Building Custom Rule C:/Users/test/source/repos/usageTest/usageTest/CMakeLists.txt
C:\Users\test\source\repos\usageTest\out\build\x64-Debug\usageTest\LINK : fatal error LNK1104: cannot open file 'pugixml.lib' 

Build All failed.
````

**Comment by JonLiu1993** [comment]
I found that the upstream find_package (pugixml 1.9 QUIET) has obvious version requirements for the pugixml port. I added find_package (pugixml CONFIG REQUIRED) to the CMakeLists to successfully find pugixml.lib

**Comment by myd7349** [comment]
Hi! @JonLiu1993 Thanks for your review.

I have created a PR to address this issue: https://github.com/xdf-modules/libxdf/pull/38.

**Review by JonLiu1993 — DISMISSED** [review]

**Comment by JonLiu1993** [comment]
Tested usage successfully by `libxdf:x64-windows`:
````
libxdf provides CMake targets:

    # this is heuristically generated, and may not be correct
    find_package(libxdf CONFIG REQUIRED)
    target_link_libraries(main PRIVATE XDF::xdf)
````

File: ports/libxdf/vcpkg.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{
+  "name": "libxdf",
+  "version-date": "2023-08-02",
```
Comment (JavierMatosD) [inline ports/libxdf/vcpkg.json:3]: It looks like the upstream version is 0.99.8. 
```suggestion
  "version": "0.99.8",
```

Comment (myd7349) [reply ports/libxdf/vcpkg.json:3]: Hi! @JavierMatosD There was a bug in `0.99.8` and have been fixed by https://github.com/xdf-modules/libxdf/pull/38. That why I `upgrade` from `0.99.8` to latest commit. Without this fix, a 60~ LOC patch file is necessary.

Comment (myd7349) [reply ports/libxdf/vcpkg.json:3]: If a stable release is preferred, then I can revert the last two commits of this PR.

**Pull request merged**